pizjuce/MiddyMorphy/MidiMorph.cpp

```diff
@@ -15,102 +14,74 @@ MidiMorph::MidiMorph()
     autoKey             = true;
     autoLearn           = true;
     controllerListWidth = 150;
-    // Bouml preserved body end 0003540D
 }
 
 MidiMorph::~MidiMorph()
 {
-    // Bouml preserved body begin 00041F8D
-    if (guiState != 0)
+    if (guiState != nullptr)
     {
         deleteAndZero (guiState);
     }
     deleteAndZero (cursor);
-    // Bouml preserved body end 00041F8D
 }
 
 void MidiMorph::setCursorXRatio (float x)
 {
-    // Bouml preserved body begin 00042D8D
-    cursor->setXY ((float) roundToInt (x * paneSize.getWidth()) - cursor->size / 2, cursor->Point::getY());
-    // Bouml preserved body end 00042D8D
+    cursor->setXY ((float) juce::roundToInt (x * paneSize.getWidth()) - cursor->size / 2, cursor->Point::getY());
 }
 
 void MidiMorph::setCursorYRatio (float y)
 {
-    // Bouml preserved body begin 00042E0D
-    cursor->setXY (cursor->Point::getX(), (float) roundToInt (y * paneSize.getHeight()) - cursor->size / 2);
-    // Bouml preserved body end 00042E0D
+    cursor->setXY (cursor->Point::getX(), (float) juce::roundToInt (y * paneSize.getHeight()) - cursor->size / 2);
 }
 
 float MidiMorph::getCursorXRatio()
 {
-    // Bouml preserved body begin 00042E8D
     return (cursor->Point::getX() + cursor->size / 2) / paneSize.getWidth();
-    // Bouml preserved body end 00042E8D
 }
 
 float MidiMorph::getCursorYRatio()
 {
-    // Bouml preserved body begin 00042F0D
     return (cursor->Point::getY() + cursor->size / 2) / paneSize.getHeight();
-    // Bouml preserved body end 00042F0D
 }
 
-//-can happen by input or slider movement.
-//
-//-if a scene is selected, (or multiple scenes) the value for that controller gets set.
-//-if no scene is selected and auto key is on, a scene gets added.
```

Comment:
Neither agree, nor disagree, just not a big fan of comments explaining code that isn't part of a lib - better structure it in a way that explains itself and needs no additional meta information ;)

But my main reasoning is that in case someone has problems to understand the function the comment would be revealed when looking in the git history (so it's not fully lost and the original intent of the original developer always accessible).
As every comment can be outdated and needs to be verified against the implementation I'd rather remove the comment and only add comments to code that A is part of a working state and B has an element of unusualness which can not be expressed by meaningful naming.

Reply:
> just not a big fan of comments explaining code that isn't part of a lib - better structure it in a way that explains itself and needs no additional meta information ;)

Sure! :-) Still, there are use cases for comments outside of libraries that I regard as very much legitimate: For example, to explain _why_ you are doing something and to explain context. On the other hand, if you need to explain _what_ you are doing, that's a bad sign; I'm with you on that one 😅

---

> add comments to code that A is part of a working state

That convinces me 👍🏻